src/linter/ui5Types/fixHints/CoreFixHintsGenerator.ts

```diff
@@ -0,0 +1,321 @@
+
+	// Migration not possible
+	// Recommended replacement only available on ManagedObject
+	// ["fireValidationError", {}],
+
+	// Migration not possible
+	// Recommended replacement only available on ManagedObject
+	// ["fireValidationSuccess", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["getApplication", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// There is a public replacement for the most common use case that checks the
+	// result for a single library (Library.isLoaded(name))
+	// ["getLoadedLibraries", {}],
+
+	// Migration not possible
+	// Different return types -> Manual migration necessary
+	// ["getMessageManager", {}],
+
+	// Migration not possible
+	// ["getModel", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["getRenderManager", {}],
+
+	// Migration not possible
+	// ["getRootComponent", {}],
+
+	// Migration not possible
+	// We can't determine whether the static UIArea is requested
+	// ["getUIArea", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["getUIDirty", {}],
+
+	// Migration not possible
+	// Recommended replacement only available on ManagedObject
+	// ["hasModel", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["includeLibraryTheme", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["isInitialized", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["isLocked", {}],
+
+	// Migration not possible
+	// Developers should migrate to the theme-applied event
+	// ["isThemeApplied", {}],
+
+	// Migration not possible
+	// API has been removed, migration likely involves more than removing the usage
+	// ["lock", {}],
+
+	// Migration not possible
+	// ["registerPlugin", {}],
+
+	// Migration not possible
+	// ["sap.ui.core.Core.extend", {}],
+
+	// Migration not possible
+	// ["sap.ui.core.Core.getMetadata", {}],
+
+	// Migration not possible
+	// ["setModel", {}],
+
+	// Migration not possible
+	// ["setRoot", {}],
+
+	// Migration not possible
+	// ["setThemeRoot", {}],
+
+	// Migration not possible
+	// ["unlock", {}],
+
+	// Migration not possible
+	// ["unregisterPlugin", {}],
+]);
+
+export default class CoreFixHintsGenerator {
+	constructor(
+		private ambientModuleCache: AmbientModuleCache
+	) {
+
+	}
+
+	getFixHints(node: ts.CallExpression | ts.AccessExpression): FixHints | undefined {
+		if (!ts.isPropertyAccessExpression(node)) {
+			return undefined;
+		}
+		const moduleName = node.expression.getText();
+		if (moduleName !== "Core" && moduleName !== "sap.ui.getCore()") {
+			return undefined;
+		}
```

Comment:
This currently limits the autofix to only those cases: Using a local variable exactly named `Core` or using `sap.ui.getCore()` (literally, as `getText` is used).

I guess with the larger refactoring of the autofix this will change anyways, but now we have the `Ui5TypeInfo` available (part of the deprecationInfo), which could be used here to check against the `sap/ui/core/Core` module.

Reply:
Thanks!